odl/discr/lp_discr.py

```diff
@@ -228,6 +234,12 @@ def exponent(self):
         """The exponent ``p`` in ``L^p``."""
         return self._exponent
 
+    @property
+    def vector_field_space(self):
+        """vector field space."""
+        self.real_space = self.astype(self._real_dtype)
+        return ProductSpace(self.real_space, self.ndim)
```

Comment:
Well that makes the naming even more of a nightmare, what would you call such a method?

Also are we sure things generalize nicely to tensors?